config/fcos/v1_6_exp/validate_test.go

```diff
@@ -231,3 +231,129 @@ func TestValidateGrubUser(t *testing.T) {
 		})
 	}
 }
+
+func TestValidateConfig(t *testing.T) {
+	tests := []struct {
+		config  Config
+		err     error
+		errPath path.ContextPath
+	}{
+		// valid config
+		{
+			config: Config{
+				Config: base.Config{
+					Storage: base.Storage{
+						Disks: []base.Disk{
+							{
+								Device:    "/dev/vda",
+								WipeTable: util.BoolToPtr(false),
+								Partitions: []base.Partition{
+									{
+										Label:  util.StrToPtr("foo"),
+										Number: 1,
+									},
+									{
+										Label:  util.StrToPtr("bar"),
+										Number: 2,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			err:     nil,
+			errPath: path.New("yaml"),
+		},
+		// valid config as wipe_tables is true
+		{
+			config: Config{
+				Config: base.Config{
+					Storage: base.Storage{
+						Disks: []base.Disk{
+							{
+								Device:    "/dev/vda",
+								WipeTable: util.BoolToPtr(true),
+								Partitions: []base.Partition{
+									{
+										Label:  util.StrToPtr("foo"),
+										Number: 0,
+									},
+									{
+										Label:  util.StrToPtr("bar"),
+										Number: 0,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			err:     nil,
+			errPath: path.New("yaml"),
+		},
+		// specify root disk
+		{
+			config: Config{
+				Config: base.Config{
+					Storage: base.Storage{
+						Disks: []base.Disk{
+							{
+								Device:    "/dev/disk/by-id/coreos-boot-disk",
+								WipeTable: util.BoolToPtr(false),
+								Partitions: []base.Partition{
+									{
+										Label:  util.StrToPtr("foo"),
+										Number: 0,
+									},
+									{
+										Label:  util.StrToPtr("bar"),
+										Number: 2,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			err:     nil,
+			errPath: path.New("yaml"),
+		},
+		// invalid config
```

Comment:
Whats invalid about it ?

Reply:
the config below had wipe_tables set to false and the 0th partition number set to 0. so i just commented it as an example of an invalid config